crates/rpc/src/handlers/state.rs

```diff
@@ -77,3 +78,46 @@ pub async fn get_state_root(state_id: ID, db: ReamDB) -> Result<impl Reply, Reje
         StatusCode::OK,
     ))
 }
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct WithdrawalData {
+    #[serde(with = "serde_utils::quoted_u64")]
+    validator_index: u64,
+    #[serde(with = "serde_utils::quoted_u64")]
+    amount: u64,
+    #[serde(with = "serde_utils::quoted_u64")]
+    withdrawable_epoch: u64,
+}
+// Called by `/states/{state_id}/get_pending_partial_withdrawals` to get pending partial withdrawals
+// for state with given stateId
+pub async fn get_pending_partial_withdrawals(
+    state_id: ID,
+    db: ReamDB,
+) -> Result<impl Reply, Rejection> {
+    let state = get_state_from_id(state_id, &db).await?;
+
+    let withdrawals = state.get_expected_withdrawals();
+    let partial_withdrawals: Vec<Withdrawal> = withdrawals
+        .into_iter()
+        .filter(|w| {
+            let validator = &state.validators[w.validator_index as usize];
+            let balance = state.balances[w.validator_index as usize];
+            validator.is_partially_withdrawable_validator(balance)
+        })
+        .collect();
+
+    let withdrawal_data: Vec<WithdrawalData> = partial_withdrawals
+        .into_iter()
+        .map(|w| WithdrawalData {
+            validator_index: w.validator_index,
+            amount: w.amount,
+            withdrawable_epoch: state.get_current_epoch(),
+        })
+        .collect();
+    let response = warp::reply::with_header(
+        BeaconVersionedResponse::json(withdrawal_data),
+        "Eth-Consensus-Version",
+        "electra",
+    );
+
+    Ok(with_status(response, StatusCode::OK))
```

Comment:
```suggestion
    Ok(with_status(with_header(
        BeaconVersionedResponse::json(withdrawal_data),
        ETH_CONSENSUS_VERSION_HEADER,
        ELECTRA,
    ), StatusCode::OK))
```

can you define these consts

Reply:
ok will do it